### ISSUE TITLE
Function signature help doesn't work, when inside other parentheses.

### ISSUE THREAD

**Issue by auronen** [kirides/DaedalusLanguageServer#8]
```c++
if (NPC_KnowInfo()) // Doesn't show correct completions
```

Fails with `langserver/lsphandler.go:98	signature completion error index out of bounds:`

**Comment by kirides** [comment]
Completions are also currently not counting in function-symbols.

e.g. 

```
func string This() { return ""; };

// ...
const string a = "a";
ConcatStrings(a, [Completing doesn't show methods])
```

**Comment by auronen** [comment]
That is a very good point, offer functions with the right return value as a completion.

**Comment by kirides** [comment]
> That is a very good point, offer functions with the right return value as a completion.

fixed as of 0efd9e66c30092bcba16e5d11972bff1805a1e13

Now its just about finally fixing the method-call context function to work properly

```
// -> '|' is the cursor in this example

// DOESNT WORK                 \/ doesn't show proper intellisense
ConcatStrings("", ConcatStrings(|))

// Works                       \/ mind the space after the cursor
ConcatStrings("", ConcatStrings(| ))
```
